jobs/generation/PRTestGenerator.groovy

```diff
@@ -0,0 +1,482 @@
+      case 'release/2.0.0':
+      case 'release/2.1':
+      case 'release/2.2':
+        return 'bin'
+      case 'release/3.0':
+      case 'master':
+        return 'artifacts/bin'
+      default:
+        assert false : "Unknown branch: '${branch}'"
+        break
+    }
+  }
+
+  String getProjectFolderName() {
+    return getProjectFolderName(name)
+  }
+
+  String getJobFolderName(String branch) {
+    return getJobFolderName(name, branch)
+  }
+
+  def static getOSGroup(def os) {
+    def osGroupMap = ['Tizen': 'Linux']
+    def osGroup = osGroupMap.get(os, null)
+    assert osGroup != null : "Could not find os group for ${os}"
+    return osGroupMap[os]
+  }
+
+  def static addArchival(def job, String filesToArchive, String filesToExclude = '') {
+    job.with {
+      publishers {
+        flexiblePublish {
+          conditionalAction {
+            condition {
+              status('SUCCESS', 'SUCCESS')
+            }
+
+            publishers {
+              archiveArtifacts {
+                allowEmpty(false)
+                pattern(filesToArchive)
+                exclude(filesToExclude)
+
+                // Always archive so that the flexible publishing
+                // handles pass/fail
+                onlyIfSuccessful(false)
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+class ProjectCoreclr extends Project {
+  def setupJob(def newJob, String branch, String scenario, boolean isPR, String architecture, String configuration, String os) {
+    Utilities.standardJobSetup(newJob, name, true, "*/${branch}")
+
+    addPRTriggers(newJob, branch, architecture, os, configuration, scenario)
+    setJobTimeout(newJob, architecture, configuration, scenario)
+
+    newJob.with {
+      steps {
+        copyArtifacts("${getJobFolderName('dotnet/corefx', branch)}/${getJobName(configuration != 'Checked' ? configuration : 'Release', architecture, os, scenario)}") {
+          includePatterns("${getCorefxBinPath(branch)}/build.tar.gz")
+          buildSelector {
+            latestSuccessful(true)
+          }
+        }
+      }
+    }
+
+    if (branch == 'release/2.0.0') {
+      String patch = 'coreclr-v2.0.0-tests.patch'
+      String newJobName = getJobName(configuration, architecture, os, scenario)
+      String folderName = getJobFolderName(branch)
+      String projectDir = Utilities.getFullJobName(name, newJobName, isPR, folderName)
+      Utilities.getPatch(newJob, patch)
+      Utilities.applyPatch(newJob, patch)
+    }
+
+    List<String> buildCommands = calculateBuildCommands(newJob, scenario, branch, architecture, configuration, os, configuration != 'Checked')
+
+    newJob.with {
+      steps {
+        buildCommands.each { buildCommand ->
+          shell(buildCommand)
+        }
+      }
+    }
+  }
+
+  boolean shouldGenerateJob(String scenario, boolean isPR, String architecture, String configuration, String os) {
+    // Generate only PR jobs
+    return isPR;
```

Comment:
Why do we generate only isPR jobs?

Reply:
For coreclr and coresetup we generate only PR jobs, as we want to test only PRs. For corefx we need to generate non PR job that  should be triggered after each push as we need latest corefx artifacts to run coreclr tests.